File: fundamentos/desafio-js/src/questions/get-client-w-highest-balance-at-ag-10.ts
import { IAccount } from "../i-account";
/** O nome do(a) cliente com o maior saldo na agência 10 é: */
export default function getClientWHighestBalanceAtAg10(...accounts: Array<IAccount>) {
  const acc = accounts
    .filter(acc => acc.agencia == 10)
    .sort((a, b) => b.balance - a.balance)[0]
  
  console.info(`O nome do(a) cliente com o maior saldo na agência 10 é: ${acc.name}, $${acc.balance}`);
}

File: fundamentos/desafio-dart/lib/questions/get_client_w_highest_balance_at_ag_10.dart
import 'package:desafio_dart/domains/account.dart';

/// O nome do(a) cliente com o maior saldo na agência 10 é:
void getClientWHighestBalanceAtAg10(List<Account> accounts) {
  final accs = accounts.where((acc) => acc.agencia == 10).toList();

  accs.sort((Account a, Account b) => b.balance.compareTo(a.balance));

  final client = accs.first;

  print(
      'O nome do(a) cliente com o maior saldo na agência 10 é: ${client.name}, \$${client.balance}');
}
